modules/repository-s3/src/internalClusterTest/java/org/elasticsearch/repositories/s3/S3BlobStoreRepositoryMetricsTests.java

```diff
@@ -166,6 +184,43 @@ public void testMetricsWithErrors() throws IOException {
         assertThat(getNumberOfMeasurements(plugin, HTTP_REQUEST_TIME_IN_MICROS_HISTOGRAM, Operation.DELETE_OBJECTS), equalTo(1L));
     }
 
+    public void testMetricsForRequestRangeNotSatisfied() throws IOException {
+        final String repository = createRepository(randomRepositoryName());
+        final String dataNodeName = internalCluster().getNodeNameThat(DiscoveryNode::canContainData);
+        final BlobContainer blobContainer = getBlobContainer(dataNodeName, repository);
+        final TestTelemetryPlugin plugin = getPlugin(dataNodeName);
+
+        final OperationPurpose purpose = randomFrom(OperationPurpose.values());
+        final String blobName = randomIdentifier();
+
+        for (int i = 0; i < randomIntBetween(1, 3); i++) {
+            final long batch = i + 1;
+            addErrorStatus(TOO_MANY_REQUESTS, TOO_MANY_REQUESTS, REQUESTED_RANGE_NOT_SATISFIED);
```

Comment:
Ideally it would be great to test multiple `REQUESTED_RANGE_NOT_SATISFIED` errors in one request, but I'm not sure how it's possible.